[mcha]feat: 기본적인 유효성 검사

charmingjae opened the pull request:
[-] argument 개수가 2개인지 검사
[-] 맵 이름이 .cub로 끝나는지 검사
[-] 파일을 오픈하여 file descriptor 값이 양수인지 검사

Commit:
[mcha]feat: 기본적인 유효성 검사
[-] argument 개수가 2개인지 검사
[-] 맵 이름이 .cub로 끝나는지 검사
[-] 파일을 오픈하여 file descriptor 값이 양수인지 검사

Commit:
[mcha]feat: gnl을 사용하여 맵 읽어오는 작업 수행
[-] gnl이 받아오는 사이즈가 0보다 클 경우에 정상적인 맵 파싱 진행
[-] 의미 있는 구문이 들어올 경우 판단하는 프로세스 차후 추가 예정

kyj93790 commented on src/parse/check/check_main_arg.c:
```diff
@@ -0,0 +1,71 @@
+	int		fd;
+
+	tmp = av[1];
+	fd = open(tmp, O_RDWR, 0644);
```
map file에 write을 할 상황이 없으니까 open 옵션은 O_RDONLY로 주는게 낫지 않을까요??

Commit:
[mcha]feat: additional parsing proc
[-] Texture component가 들어왔을 때 유효한지 판단
[-] Texture component 타입을 비교하여 할당 과정 추가 예정

charmingjae committed:
[mcha]feat: texture 파싱 처리
[-] 문자열 파싱 후 이차원 배열의 사이즈가 2인 경우에만 진행
[-] 이차원 배열의 0번째 인덱스의 문자열을 비교하여 일치하는 경우에만 진행

Commit:
[mcha]feat: texture, color 파싱 과정 추가
[-] map 파싱 부분 추가 예정

Commit:
[mcha]feat: 맵 파싱 마무리
[-] 예외사항 추가적으로 생각 해보기

charmingjae committed:
[mcha]feat: opengl 준비

Commit:
[mcha]feat: minilibx, image structure
[-] minilibx 추가
[-] 이미지 사용을 위한 구조체 추가
